File: eTheater.Model/Purchase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eTheater.Model
{
    public class Purchase
    {
        public int PurchaseId { get; set; }

        public int? TotalPrice { get; set; }

        public int? NumberOfTickets { get; set; }

        public bool IsPaid { get; set; }

        public string? PaymentIntentId { get; set; }

        public int? UserId { get; set; }

        public virtual User? User { get; set; }

        public int? ShowScheduleId { get; set; }

        public virtual ShowSchedule? ShowSchedule { get; set; }

        public DateTime CreatedAt { get; set; }
    }
}

File: UI/etheater_mobile/lib/models/purchase.dart
import 'package:json_annotation/json_annotation.dart';
import 'models.dart';

part 'purchase.g.dart';

@JsonSerializable()
class Purchase {
  int purchaseId;
  int totalPrice;
  int numberOfTickets;
  bool isPaid;
  String? paymentIntentId;
  int? userId;
  User user;
  int? showScheduleId;
  ShowSchedule showSchedule;
  DateTime createdAt;

  Purchase(
      {required this.purchaseId,
      required this.totalPrice,
      required this.numberOfTickets,
      required this.isPaid,
      this.paymentIntentId,
      this.userId,
      required this.user,
      this.showScheduleId,
      required this.showSchedule,
      required this.createdAt});

  factory Purchase.fromJson(Map<String, dynamic> json) =>
      _$PurchaseFromJson(json);
  Map<String, dynamic> toJson() => _$PurchaseToJson(this);
}
